src/Uno.UI/UI/Xaml/Controls/CalendarDatePicker/CalendarDatePicker.Android.cs

```diff
@@ -0,0 +1,205 @@
+﻿#if XAMARIN_ANDROID
+using Uno.Extensions;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Automation.Peers;
+using Windows.UI.Xaml.Controls.Primitives;
+
+namespace Windows.UI.Xaml.Controls
+{
+
+
+	public partial class CalendarDatePicker : Control
+	{
+		public event EventHandler<CalendarDatePickerDateChangedEventArgs> DateChanged;
+
+		public CalendarDatePicker()
+		{
+			InitializeVisualStates();
+		}
+
+		#region DateProperty
+
+		public Nullable<DateTimeOffset> Date
+		{
+			get { return (DateTimeOffset?)this.GetValue(DateProperty); }
+			set { this.SetValue(DateProperty, value); }
+		}
+
+		private string _DateString;
+
+		//#18331 If the Date property of DatePickerFlyout is two way binded, the ViewModel receives the control's default value while the ViewModel sends its default value which desynchronizes the values
+		//Set initial value of DatePicker to DateTimeOffset.MinValue to avoid 2 way binding issue where the DatePicker reset Date(DateTimeOffset.MinValue) after the initial binding value.
+		//We assume that this is the view model who will set the initial value just the time to fix #18331
+
+		public static DependencyProperty DateProperty { get; } =
+	DependencyProperty.Register("Date", typeof(DateTimeOffset?), typeof(CalendarDatePicker), new PropertyMetadata(DateTimeOffset.MinValue,
+		(s, e) => ((CalendarDatePicker)s).OnDatePropertyChanged((DateTimeOffset?)e.NewValue, (DateTimeOffset?)e.OldValue)));
+
+		private void OnDatePropertyChanged(DateTimeOffset? newValue, DateTimeOffset? oldValue)
+		{
+
+			string dateFormatString = "d";
+			if (this.DateFormat != default(string))
+			{
+				dateFormatString = this.DateFormat;
+			}
+
+			if (!newValue.HasValue)
+			{
+				_DateString = this.PlaceholderText;
+			}
+			else
+			{
+				_DateString = newValue.Value.ToString(dateFormatString);
+			}
+
+			DateChanged?.Invoke(this, new CalendarDatePickerDateChangedEventArgs(newValue, oldValue));
+		}
+
+		#endregion
+
+		#region PlaceholderText
+
+
+		public string PlaceholderText
+		{
+			get { return (string)this.GetValue(PlaceholderTextProperty); }
+			set { this.SetValue(PlaceholderTextProperty, value); }
+		}
+
+		public static global::Windows.UI.Xaml.DependencyProperty PlaceholderTextProperty { get; } =
+		Windows.UI.Xaml.DependencyProperty.Register(
+			"PlaceholderText", typeof(string),
+			typeof(global::Windows.UI.Xaml.Controls.CalendarDatePicker),
+			new FrameworkPropertyMetadata(default(string)));
+
+		#endregion
+
+
+		#region DateFormat
+		public string DateFormat
+		{
+			get { return (string)this.GetValue(DateFormatProperty); }
+			set { this.SetValue(DateFormatProperty, value); }
+		}
+
+		public static global::Windows.UI.Xaml.DependencyProperty DateFormatProperty { get; } =
+		Windows.UI.Xaml.DependencyProperty.Register(
+			"DateFormat", typeof(string),
```

Comment:
Can you validate that windows really has null as a default value for this property?

Reply:
It has empty string.
Changing.